File: 543 - Diameter of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let ans = 0;
var diameterOfBinaryTree = function(root) {
    ans = 0;
    depth(root);
    return ans;
};
function depth(root) {
    if (!root) {
        return 0;
    }
    let l, r;
    l = depth(root.left);
    r = depth(root.right);
    ans = Math.max(l + r, ans);
    return Math.max(l, r) + 1;
}

File: 543 - Diameter of Binary Tree.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */
class Solution {
    int ans = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        this.depth(root);
        return this.ans;
    }
    public int depth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int l, r;
        l = depth(root.left);
        r = depth(root.right);
        ans = Math.max(ans, l + r);
        return Math.max(l, r) + 1;

    }
}